TELCODOCS-1506 Telco CORE Reference Design Specification (#65732)
openshift/openshift-docs: modules/telco-core-storage.adoc

+
OpenShift Data Foundation (ODF) is a Ceph based software-defined storage solution for containers. It provides block storage, file system storage, and on-premises object storage, which can be dynamically provisioned for both persistent and non-persistent data requirements. Telco core applications require persistent storage.
+
It is important to note that all storage data may not be encrypted in flight. To reduce risk the storage network should be isolated from other cluster networks. The storage network must not be reachable (routable) from other cluster networks, that is only nodes directly attached to the storage network can gain access to it.

== OpenShift Data Foundation

New in this release::

Not applicable

Description::

OpenShift Data Foundation (ODF) is a software-defined storage service for containers. For telco core clusters storage support is provided by external (storage services running outside the application workload cluster) ODF.

* ODF supports segregation of storage traffic using secondary CNI networks.

Limits and requirements::

> * ODF when running on dual-stack uses IPv4 addressing. For more information, see [https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/latest/html-single/4.13_release_notes/index#support_openshift_dual_stack_with_odf_using_ipv4[Support OpenShift dual stack with ODF using IPv4].

Reviewer: Flip the sentence subject/object order for clarity.

```suggestion
* In an IPv4/IPv6 dual-stack networking environment, {rh-storage} uses IPv4 addressing.
For more information, see [https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/latest/html-single/4.13_release_notes/index#support_openshift_dual_stack_with_odf_using_ipv4[Support OpenShift dual stack with ODF using IPv4].
```

Engineering considerations::

> * ODF network traffic should be isolated from other traffic on a dedicated network for example VLAN isolation.

Reviewer: ```suggestion
* {rh-storage} network traffic should be isolated from other traffic on a dedicated network. For example, by using VLAN isolation.
```

== Other Storage

> Other storage solutions may be used to provide persistent storage for core clusters. The configuration and integration of these solutions is outside the scope of the reference design specification (RDS). Integration of the storage solution into the core cluster must include correct sizing and performance analysis to ensure the storage meets overall performance and resource utilization requirements.

Reviewer: ```suggestion
Other storage solutions may be used to provide persistent storage for core clusters. The configuration and integration of these solutions is outside the scope of the telco core reference design specification. Integration of the storage solution into the core cluster must include correct sizing and performance analysis to ensure the storage meets overall performance and resource utilization requirements.
```

Reviewer: RDS doesn't need an initialism for external audiences.

// Module included in the following assemblies:
//
// * telco_ref_design_specs/core/telco-core-ref-components.adoc

:_mod-docs-content-type: REFERENCE
[id="telco-core-storage_{context}"]
= Storage

Overview::
Cloud native storage services can be provided by multiple solutions including {rh-storage} from Red Hat or third parties.
+
{rh-storage} is a Ceph based software-defined storage solution for containers. It provides block storage, file system storage, and on-premises object storage, which can be dynamically provisioned for both persistent and non-persistent data requirements. Telco core applications require persistent storage.
+

> It is important to note that all storage data may not be encrypted in flight. To reduce risk the storage network should be isolated from other cluster networks. The storage network must not be reachable (routable) from other cluster networks, that is only nodes directly attached to the storage network can gain access to it.

Reviewer 2: ```suggestion
[NOTE]
====
All storage data may not be encrypted in flight. To reduce risk, isolate the storage network from other cluster networks. The storage network must not be reachable, or routable, from other cluster networks. Only nodes directly attached to the storage network should be allowed to gain access to it.
====
```

== {rh-storage}

New in this release::

Not applicable

Description::

{rh-storage-first} is a software-defined storage service for containers.
For telco core clusters, storage support is provided by {rh-storage} storage services running externally to the application workload cluster.

> * {rh-storage} supports separation of storage traffic using secondary CNI networks.

Reviewer 2: I would add this as a sentence to the previous paragraph - why is it a separate bullet?

Limits and requirements::
* In an IPv4/IPv6 dual-stack networking environment, {rh-storage} uses IPv4 addressing. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.13/html-single/4.13_release_notes/index#support_openshift_dual_stack_with_odf_using_ipv4[Support OpenShift dual stack with ODF using IPv4].

Engineering considerations::

> * {rh-storage} network traffic should be isolated from other traffic on a dedicated network. For example, by using VLAN isolation.

Reviewer 2: ```suggestion
* {rh-storage} network traffic should be isolated from other traffic on a dedicated network, for example, by using VLAN isolation.
```

== Other Storage

> Other storage solutions may be used to provide persistent storage for core clusters. The configuration and integration of these solutions is outside the scope of the telco core reference design specification. Integration of the storage solution into the core cluster must include correct sizing and performance analysis to ensure the storage meets overall performance and resource utilization requirements.

Reviewer 2: ```suggestion
Other storage solutions can be used to provide persistent storage for core clusters. The configuration and integration of these solutions is outside the scope of the telco core reference design specification. Integration of the storage solution into the core cluster must include correct sizing and performance analysis to ensure the storage meets overall performance and resource utilization requirements.
```
Per IBM Style: Use "may" only to indicate permission. Otherwise, use "can" to indicate ability, or use "might" to indicate possibility. "May" is overused in technical information. In most cases, use "can" or "might" instead.
